docs/integrations/creating-nodes/build/n8n-node.md

```diff
@@ -0,0 +1,186 @@
+    * **OAuth2**: Use an OAuth 2.0 flow to get access tokens on behalf of a user or app.
+    * **Basic Auth**: Send the base64-encoded username and password through Authorization headers.
+    * **Custom**: Create your own credential logic. This will create an empty credential class that you can customize according to your needs.
+    * **None**: No authentication necessary. Don't create a credential class for the node.
+
+Once you've made your selections, `n8n-node` will create a new project directory for your node in the current directory. By default, it will also install the initial project dependencies (you can disable this by passing the `--skip-install` flag).
+
+### Providing node details on the command line
+
+You can provide some of your node details on the command line to avoid prompts.
+
+You can include the name you want to use for your node an argument:
+
+```shell
+n8n-node new n8n-nodes-myproject
+```
+
+/// note | Node name format
+Node names must use one of the following formats:
+
+* `@<YOUR_ORG>/n8n-nodes-<YOUR_NODE_NAME>`
+* `n8n-nodes-<YOUR_NODE_NAME>`
+///
+
+If you know the template you want to use ahead of time, you can also pass the value using the `--template` flag:
+
+```shell
+n8n-node new --template declarative/custom
+```
+
+The template must be one of the following:
+
+* `declarative/github-issues`: A demo node that includes multiple operations and credentials. This can help you get familiar with the node structure and conventions.
+* `declarative/custom`: A blank template that will guide you through your custom setup with some further prompts.
+* `programmatic/example`: A programmatic style node with full flexibility.
+
+## Building your node
+
+You can build your node by running the `build` command in your project's root directory:
+
+```shell
+n8n-node build
+```
+
+`n8n-node` will compile your TypeScript files and bundle your other project assets. You can also call the `build` script from your package manager. For instance, if you're using `npm`, this works the same:
+
+```shell
+npm run build
+```
+
+## Lint your node
+
+The `n8n-node` tool automatically creates a `lint` script for your project as well. You can run with your package manager. For example:
+
+```shell
+npm run lint
+```
+
+## Testing your node in n8n
+
+To test your node in n8n, you run the `dev` command in your project's root directory:
+
+```shell
+n8n-node dev
+```
+
+As with the `build` command, you can also run this through your package manager. For example:
+
+```shell
+npm run dev
+```
+
+`n8n-node` will compile your project and start your local `npm`-installed n8n instance with your node loaded.
+
+Visit your `localhost:5678` to sign in to your n8n instance. If you open a workflow, your node appears in the nodes panel:
+
+![node in nodes panel](/_images/integrations/creating-nodes/n8n-node/node_in_nodes_panel.png)
+
+From there, you can add it to your workflow and test the node's functionality as you develop.
```

Comment:
There is a new release command, you can find some details in the README: https://www.npmjs.com/package/@n8n/node-cli?activeTab=readme